### PR TITLE
[flutter_tools] cleanup build_tests to reduce mocking and globals usage

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

* Removes MockPlatform in favor of FakePlatform
* Removes MockProcessManager in favor of FakeProcessManager
* Updates test to consistent style of setUp declared FileSystem, reducing globals
* minor formatting changes

### CODE REVIEW

File: packages/flutter_tools/test/commands.shard/hermetic/build_fuchsia_test.dart [hunk 1]
```diff
@@ -16,104 +17,106 @@ import 'package:flutter_tools/src/project.dart';
...
 import 'package:mockito/mockito.dart';
 import 'package:process/process.dart';
-import 'package:flutter_tools/src/globals.dart' as globals;
 
 import '../../src/common.dart';
 import '../../src/context.dart';
 import '../../src/mocks.dart';
 
+// Defined globally for mocks to use.
+FileSystem fileSystem;
```
Comment (jmagman) [inline packages/flutter_tools/test/commands.shard/hermetic/build_fuchsia_test.dart]: Should it be `kFileSystem`?

**Review by jmagman — APPROVED** [review]
LGTM with naming nit

File: packages/flutter_tools/test/commands.shard/hermetic/build_fuchsia_test.dart (hunk 1 above)
Comment (jonahwilliams) [reply packages/flutter_tools/test/commands.shard/hermetic/build_fuchsia_test.dart]: but Its not a konstant

Comment (jmagman) [reply packages/flutter_tools/test/commands.shard/hermetic/build_fuchsia_test.dart]: I thought there was some naming convention for globals that conflated constants and globals but I just misread.  It's all so Hungarian...

**Pull request merged**